xbmc/input/WindowTranslator.cpp

```diff
@@ -278,18 +281,32 @@ int CWindowTranslator::GetVirtualWindow(int windowId)
     // check if we're in a DVD menu
     if (g_application.GetAppPlayer().IsInMenu())
       return WINDOW_VIDEO_MENU;
-    // check for LiveTV and switch to it's virtual window
-    else if (CServiceBroker::GetPVRManager().IsStarted() && g_application.CurrentFileItem().HasPVRChannelInfoTag())
-      return WINDOW_FULLSCREEN_LIVETV;
+    // special casing for Live TV
+    else if (g_application.CurrentFileItem().HasPVRChannelInfoTag())
+    {
+      if (g_infoManager.IsPlayerChannelInputActive())
```

Comment:
imo CWindowTranslator should not depend on GUIInfoManager. GUIInfoManager is a cache and proxy for gui controls.

Reply:
Okay, then we need to keep the pvr dependency in this method. Would that be acceptable?

Reply:
Yes, imo better than gui

Reply:
Okay, will change that.